src/pages/index.js

```diff
@@ -64,22 +102,100 @@ function getCard(cardData) {
         cardImagePopup.open(item);
       },
     },
+    handleDeleteCardClick,
+    handleLikeCardClick,
     elementSelector.cardTemplate
   );
   return cardElement.generateCard();
 }
 
+function handleLikeCardClick(card) {
+  if (card.isLiked) {
+    api
+      .removeLike(card.getCardId())
+      .then((res) => {
+        card.handleLikeButton(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err); // log the error to the console
+      });
+  } else {
+    api
+      .addLike(card.getCardId())
+      .then((res) => {
+        card.handleLikeButton(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err); // log the error to the console
+      });
+  }
+}
+
+function handleDeleteCardClick(card) {
+  cardDeleteConfirmation.open();
+  cardDeleteConfirmation.handleDeleteCardConfirmation(() => {
+    api
+      .deleteCard(card.getCardId())
+      .then((res) => {
+        card.handleDeleteButton();
+        cardDeleteConfirmation.close();
+      })
+      .catch((err) => {
+        console.error(err); // log the error to the console
+      });
+  });
+}
+
 function handleProfileFormSubmit(userData) {
-  userInfo.setUserInfo(userData);
-  profileEditFormPopup.close();
+  profileEditFormPopup.renderSaving(true);
+  api
+    .editProfile(userData)
+    .then((userData) => {
+      userInfo.setUserInfo(userData);
+      profileEditFormPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditFormPopup.renderSaving(false);
+    });
 }
 
 function handleAddCardFormSubmit(cardData) {
-  const name = cardData.title;
-  const link = cardData.link;
-  createCard({ name, link });
-  cardAddFormPopup.close();
-  cardAddForm.reset();
+  cardAddFormPopup.renderSaving(true);
+  api
+    .addCard(cardData)
+    .then((cardData) => {
+      const name = cardData.name;
+      const link = cardData.link;
+      const _id = cardData._id;
+      createCard({ name, link, _id });
+      cardAddFormPopup.close();
+      cardAddForm.reset();
+    })
+    .catch((err) => {
+      console.error(err); // log the error to the console
+    })
+    .finally(() => {
+      cardAddFormPopup.renderSaving(false);
+    });
+}
+
+function handleUpdateAvatarFormSubmit(avatar) {
+  editAvatarPopup.renderSaving(true);
+  api
+    .updateAvatar(avatar.name)
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err); // log the error to the console
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors